server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPServerFactory.java

```diff
@@ -45,31 +45,30 @@ public class IMAPServerFactory extends AbstractServerFactory {
     protected final ThrowingFunction<HierarchicalConfiguration<ImmutableNode>, ImapSuite> imapSuiteProvider;
     protected final ImapMetrics imapMetrics;
     protected final GaugeRegistry gaugeRegistry;
-    protected final ThrowingFunction<HierarchicalConfiguration<ImmutableNode>, Set<ConnectionCheck>> connectionCheckProvider;
+    protected final Set<ConnectionCheck> connectionChecks;
 
     @Inject
     @Deprecated
     public IMAPServerFactory(FileSystem fileSystem, ImapDecoder decoder, ImapEncoder encoder, ImapProcessor processor,
-                             MetricFactory metricFactory, GaugeRegistry gaugeRegistry, ThrowingFunction<HierarchicalConfiguration<ImmutableNode>, Set<ConnectionCheck>> connectionCheckProvider) {
+                             MetricFactory metricFactory, GaugeRegistry gaugeRegistry, Set<ConnectionCheck> connectionChecks) {

```

Comment:
No It was better before: one set of connections set per ports. More flexibility.